controllers/pattern_controller.go

```diff
@@ -514,3 +530,103 @@ func (r *PatternReconciler) actionPerformed(p *api.Pattern, reason string, err e
 	}
 	return r.onReconcileErrorWithRequeue(p, reason, err, nil)
 }
+
+// applyPatternAppDetails retrieves the status for the Pattern applications created in ArgoCD.
+// Once the Application Status is retrieve it appends the information
+// to the array and the CR is updated.
+func (r *PatternReconciler) applyPatternAppDetails(argoClient argoclient.Interface, input *api.Pattern) (*api.Pattern, error) {
+	output := input.DeepCopy()
+	var applicationInfo api.PatternApplicationInfo
+	var labelFilter = "validatedpatterns.io/pattern=" + output.ObjectMeta.Name
+
+	// Retrieving all applications that contain the label
+	// oc get Applications -A -l validatedpatterns.io/pattern=<pattern-name>
+	//
+	// The VP framework adds the label to each application it creates.
+	applications, err := argoClient.ArgoprojV1alpha1().Applications("").List(context.Background(),
+		metav1.ListOptions{
+			LabelSelector: labelFilter,
+		})
+	if err != nil {
+		return nil, err
+	}
+
+	// Reset the array
+	output.Status.Applications = nil
+
+	// Loop through the Pattern Applications and append the details to the Applications array
+	for _, app := range applications.Items {
+		// Add Application information to ApplicationInfo struct
+		applicationInfo.Name = app.Name
+		applicationInfo.Namespace = app.Namespace
+		applicationInfo.AppHealthStatus = string(app.Status.Health.Status)
+		applicationInfo.AppHealthMessage = app.Status.Health.Message
+		applicationInfo.AppSyncStatus = string(app.Status.Sync.Status)
+
+		// Now let's append the Application Information
+		output.Status.Applications = append(output.Status.Applications, *applicationInfo.DeepCopy())
+	}
+	return output, err
+}
+
+// updatePatternCRDetails compares the current CR Status.Applications array
+// against the instance.Status.Applications array.
+// Returns true if the CR was updated else it returns false
+func (r *PatternReconciler) updatePatternCRDetails(input *api.Pattern) (bool, error) {
+	fUpdateCR := false
+
+	// Update CR for the Pattern Instance
+	// Create an empty Pattern CR
+	existingPatternCR := &api.Pattern{}
+
+	err := r.Client.Get(context.TODO(), client.ObjectKey{
+		Namespace: "openshift-operators",
+		Name:      input.ObjectMeta.Name}, existingPatternCR)
+
+	if err != nil {
+		return false, err
+	}
```

Comment:
I think we'd want to do a deep copy of `input.Status.Applications` instead. Much cheaper than an API lookup.
This might mean combining `updatePatternCRDetails()` and `applyPatternAppDetails()` so that you can copy from the original before you update it.